Advanced search: Refactor the code

canonici opened the pull request.

Commit:
Advanced search: Factorize `event_search`

Commit:
Advanced search: Replace `~values` by `~places`

Commit:
Advanced search: Name parameters in `do_compare`

Commit:
Advanced search: Not inline pattern matching of `Date.cdate_to_dmy_opt`

Commit:
Advanced search: Use `Date.dmy` directly instead of `Date.date`

Commit:
Advanced search: Convert sex string to sex type

Commit:
gwdb: Add some comments

Commit:
Advanced search: Remove an `@@`

hgouraud commented:
where do we get Calendars.make?

michelmno commented:
> where do we get Calendars.make?

need Calendars `calendar-kind` branch as per old issue #1469

ilankri committed:
Advanced search: Simplify some code

ilankri committed:
Advanced search: Add a comment

ilankri committed:
Advanced search: Reorder some pattern matching cases

ilankri committed:
Advanced search: Add param. to prepare move out of enclosing function

ilankri committed:
Advanced search: Move some code to a separate module

ilankri committed:
Advanced search: Move some code to a separate module

ilankri committed:
Advanced search: Factorize some code

Using the following identity:

```
if condition1 then
    if condition2 then expression1 else expression2
else
    if condition3 then expression1 else expression2
=
let condition = if condition1 then condition2 else condition3 in
if condition then expression1 else expression2
```

ilankri committed:
Advanced search: Share some code

ilankri committed:
Advanced search: Share some code

ilankri committed:
Advanced search: Change the way exact places are handled

ilankri committed:
Advanced search: Inline a function

ilankri committed:
Advanced search: Inline a function

ilankri committed:
Advanced search: Change the type `AdvancedSearchMatch.match_date`

ilankri committed:
Advanced search: Check once if a person is visible

ilankri committed:
Advanced search: Let function callers get values from the environment

ilankri committed:
Advanced search: Move some code to a separate function

ilankri committed:
Advanced search: Move some code to a separate function

ilankri committed:
Advanced search: Simplify some code

ilankri committed:
Advanced search: Rename some identifiers

ilankri committed:
Advanced search: Label some function parameters

ilankri committed:
Advanced search: Inline some code

ilankri committed:
Advanced search: Rewrite some code to prepare upcoming change

ilankri committed:
Advanced search: Add a wrapper function for exact places

ilankri committed:
Advanced search: Adjust the type of `match_date`

ilankri committed:
Advanced search: Move some code

ilankri committed:
Advanced search: Reorder some function parameters

ilankri committed:
Advanced search: Name an anonymous function

ilankri committed:
Advanced search: Perform some η-reductions

ilankri committed:
Advanced search: Inline some code

ilankri committed:
Advanced search: Reorder some function arguments

ilankri committed:
Advanced search: Reverse a conditional

ilankri committed:
Advanced search: Use a pattern matching instead of a conditional

ilankri committed:
Advanced search: Move some code

ilankri committed:
Advanced search: Define a type for the search type

ilankri committed:
Advanced search: Add some module types

ilankri committed:
Advanced search: Factorize some code

ilankri committed:
Advanced search: Enumerate all cases in pattern matching expressions

ilankri committed:
gwdb: Move the definition of an identifier closer to its use

ilankri committed:
Advanced search: Reverse a conditional

ilankri committed:
Advanced search: Reorder the cases of a pattern matching

ilankri committed:
Advanced search: Distinguish between search types using pattern matching

ilankri committed:
Advanced search: Duplicate some code to prepare upcoming change

ilankri committed:
Advanced search: Define a type alias for search field names

ilankri committed:
Advanced search: Rename some functions

ilankri commented:
@canonici Could you review this PR please?
